README.md

```diff
@@ -1,30 +1,145 @@
 # rav1d
 
-**rav1d** is an **AV1** cross-platform **d**ecoder, open-source, and focused on speed and correctness. It is a Rust port of [dav1d](https://code.videolan.org/videolan/dav1d).
+**rav1d** is an AV1 cross-platform decoder, open-source, and focused on speed
+and correctness. It is a Rust port of
+[dav1d](https://code.videolan.org/videolan/dav1d).
 
-rav1d is currently experimental. Core functionality has been transpiled using [c2rust](https://github.com/immunant/c2rust), but not everything has been ported yet and the transpiled code needs to be cleaned up from unsafe transpiled Rust to safe, idiomatic Rust.
+# Building
+
+rav1d is written in Rust and uses the standard Rust toolchain to build. The Rust
+toolchain can be installed by going to https://rustup.rs. The rav1d library
+builds on stable Rust for x86, x86_64, and aarch64, but currently requires a
+nightly compiler for armv7 and riscv. The project is configured to use a nightly
+compiler by default via `rust-toolchain.toml`, but a stable build can be made
+with the `+stable` cargo flag.
+
+For x86 targets, you'll also need to install [`nasm`](https://nasm.us/) in order
+to build with assembly support.
+
+A release build can then be made using cargo:
+
+```sh
+cargo build --release
+```
+
+For development purposes you may also want to use the `opt-dev` profile, which
+runs faster than a regular debug build but has all debug checks still enabled:
+
+```sh
+cargo build --profile opt-dev
+```
+
+To build just librav1d using a stable compiler:
+
+```sh
+cargo +stable build --lib --release
+```
+
+## Feature Flags
+
+The following feature flags are supported:
+
+* `asm` - Enables optimized assembly routines, if available for the target
+  platform.
+* `bitdepth_8` - Enables support for 8 bitdepth decoding.
+* `bitdepth_16` - Enables support for 10 and 12 bitdepth decoding.
+
+All of these features are enabled by default. In order to build a version of
+librav1d that disables one or more of these features use the
+`--no-default-features` flag in combination with the `--features` flag to enable
+any desired features. For example, to build without assembly routines, which is
+useful when testing the Rust fallback functions, do the following:
+
+```sh
+cargo build --no-default-features --features="bitdepth_8,bitdepth_16"
+```
+
+## Cross-Compiling
+
+rav1d can be cross-compiled for a target other than the host platform using the
+Cargo `--target` flag. This will require passing additional arguments to the
+Rust compiler to tell it what linker to use. This can be done by setting the
```

Comment:
```suggestion
`cargo` `--target` flag. This will require passing additional arguments to
`rustc` to tell it what linker to use. This can be done by setting the
```